### DISCUSSION TITLE
How to return a Python list of Union type pyclass objects?

### DISCUSSION THREAD

**Discussion by tushushu** [PyO3/pyo3#2373] [Questions] [answered]
Hi there,
I would like to implement a `read_from_file` function as below:
```Rust
#[pyclass]
pub struct Foo {
    ...
}

#[pyclass]
pub struct Bar {
    ...
}


// Foo and Bar are all pyclasses
enum GenericType {
    FooType(Foo),
    BarType(Bar),
}

#[pyclass]
pub struct GenericWrapper{
    inner: GenericType,
}

#[pyfunction]
pub fn read_from_file(path: &str) -> Vec<GenericWrapper> {
    ...
}
```

```Python3
_arr = read_from_file(path="somewhere")
arr = [x.extract_as_foo_or_bar() for x in _arr]
```
It reads contents from a file and return a list of pyclass objects. I tried to use `Enum` as a wrapper, but I am not sure how to implement the `extract_as_foo_or_bar` part. Is there any suggestion? Thanks very much~

**Comment by birkenfeld** [comment, marked as answer]
First question - do you need the `GenericWrapper` at all? You could just return `Vec<PyObject>` from `read_from_file`...

**Reply by tushushu** [reply]
@birkenfeld It does not have to be a `GenericWrapper`, I just want to find a way to return a Vec of different pyclasses.

**Reply by tushushu** [reply]
I use the codes below and it works, thanks very much @birkenfeld 
```Rust
let mut result: Vec<PyObject> = Vec::new();
result.push(a_foo.into_py(py));
result.push(a_bar.into_py(py));
```
